Framework/PythonInterface/plugins/algorithms/CalculateEfficiencyCorrection.py

```diff
@@ -0,0 +1,256 @@
+
+    def seeAlso(self):
+        return [ "He3TubeEfficiency", "CalculateSampleTransmission",
+                 "DetectorEfficiencyCor", "DetectorEfficiencyCorUser",
+                 "CalculateEfficiency", "ComputeCalibrationCoefVan" ]
+
+    def PyInit(self):
+        self.declareProperty(
+            MatrixWorkspaceProperty('InputWorkspace', '',
+                                    direction=Direction.Input,
+                                    optional=PropertyMode.Optional,
+                                    validator=CommonBinsValidator()),
+            doc='Input workspace with wavelength range to calculate for the correction.')
+
+        self.declareProperty(name='WavelengthRange', defaultValue='',
+                             doc='Wavelength range to calculate efficiency for.')
+
+        self.declareProperty(
+            MatrixWorkspaceProperty('OutputWorkspace', '',
+                                    direction=Direction.Output),
+            doc="Outputs the efficiency correction which can be applied by multiplying \
+                 the OutputWorkspace to the workspace that requires the correction.")
+
+        self.declareProperty(
+            name='ChemicalFormula', defaultValue='None',
+            validator=StringMandatoryValidator(),
+            doc='Sample chemical formula used to determine :math:`\\sigma (\\lambda_{ref})` term.')
+
+        self.declareProperty(
+            name='DensityType', defaultValue = 'Mass Density',
+            validator=StringListValidator(['Mass Density', 'Number Density']),
+            doc = 'Use of Mass density or Number density')
+
+        self.declareProperty(
+            name='Density',
+            defaultValue=0.0,
+            validator=FloatBoundedValidator(0.0),
+            doc='Mass density (g/cm^3) or Number density (atoms/Angstrom^3), :math:`\\rho`. \
+                 Default=0.0')
+
+        self.declareProperty(
+            name='Thickness',
+            defaultValue=1.0,
+            validator=FloatBoundedValidator(0.0),
+            doc='Sample thickness (cm), :math:`T`. Default value=1.0')
+
+        self.declareProperty(
+            name='MeasuredEfficiency',
+            defaultValue=0.0,
+            validator=FloatBoundedValidator(0.0, 1.0),
+            doc="Directly input the efficiency, :math:`\\epsilon`, measured at \
+                 MeasuredEfficiencyWavelength, :math:`\\lambda_{\\epsilon}`, to determine :math:`\\rho * T`, where \
+                 :math:`\\rho * T = - ln(1-\\epsilon) \\frac{1}{ \\frac{\\lambda_{\\epsilon} \\sigma (\\lambda_{ref})}{\\lambda_{ref}}}` \
+                 term for XSectionType == AttenuationXSection and \
+                 :math:`\\rho * T = - ln(1-\\epsilon) \\frac{1} \
+                    { \\sigma_s + \\frac{\\lambda_{\\epsilon} \\sigma (\\lambda_{ref})}{\\lambda_{ref}}}` \
+                 for XSectionType == TotalXSection \
+                 with :math:`\\lambda_{ref} =` 1.7982. Default=0.0")
```

Comment:
This tooltip doc string does not render nicely as a tooltip in my system. I think the problem is with the `\` at line ends.

Further, the default value is automatically show in the html documentation. On some systems, the tooltip text in the automatically generate dialog is cut at the first dot `.`. This i I recommend either dropping the 'Default=0.0'.

Reply:
I'll probably just remove the defaults.